main/js/download-scritps.js

```diff
@@ -215,4 +298,120 @@ function activateTheButton(os, arch){
+
+  document.getElementById("winarm-installer-link").href = buildMatrix.windows.arm64Installer.url;
+  document.getElementById("winx64-installer-link").href = buildMatrix.windows.x64Installer.url;
+  document.getElementById("winx86-installer-link").href = buildMatrix.windows.x86Installer.url;
+  document.getElementById("winarm-pack-link").href = buildMatrix.windows.arm64.url;
+  document.getElementById("winx64-pack-link").href = buildMatrix.windows.x64.url;
+  document.getElementById("winx86-pack-link").href = buildMatrix.windows.x86.url;
+
+
+  document.getElementById("macuni-version").textContent = buildMatrix.macos.installer.version;
+  document.getElementById("macapplesilicon-version").textContent = buildMatrix.macos.appleSilicon.version;
+  document.getElementById("macintel-version").textContent = buildMatrix.macos.intel.version;
+
+  document.getElementById("macuni-link").href = buildMatrix.macos.installer.url;
+  document.getElementById("macapplesilicon-link").href = buildMatrix.macos.appleSilicon.url;
+  document.getElementById("macintel-link").href = buildMatrix.macos.intel.url;
+
+
+  document.getElementById("linuxarm-appimage-version").textContent = buildMatrix.linux.arm64Installer.version;
+  document.getElementById("linuxx64-appimage-version").textContent = buildMatrix.linux.x64Installer.version;
+  document.getElementById("linuxarm-binaries-version").textContent = buildMatrix.linux.arm64.version;
+  document.getElementById("linuxx64-binaries-version").textContent = buildMatrix.linux.x64.version;
+
+  document.getElementById("linuxarm-appimage-link").href = buildMatrix.linux.arm64Installer.url;
+  document.getElementById("linuxx64-appimage-link").href = buildMatrix.linux.x64Installer.url;
+  document.getElementById("linuxarm-binaries-link").href = buildMatrix.linux.arm64.url;
+  document.getElementById("linuxx64-binaries-link").href = buildMatrix.linux.x64.url;
+
+    if (populateAutoUpdate){
+      activateTheButton();
+    }
+  }
+
+function get_info(response){
+  console.log(response);
+
+  if (!response || !response.hasOwnProperty("assets")){
+
+    console.log("Early return. Response is null or does not have assets");
+    return;
+  }
+
+  let newVersion = fallbackVersion;
+
+  if (response.hasOwnProperty("tag_name")){
+    // Our tag names have the v, but in a lot of places we don't need it, so cut it off.
+    if (response.tag_name[0] === `v`){
+      newVersion = response.tag_name.slice(-response.tag_name.length + 1);
+    } else {
+      newVersion = response.tag_name;
+    }
+  }
+
+  // return if we already have this version.
+  if (newVersion === fallbackVersion){
+    return;
+  }
+
+  let x = 0;
+
+  while (x < response.assets.length){
+    if (!response.assets[x].hasOwnProperty("name") || !response.assets[x].hasOwnProperty("browser_download_url")){
+      continue;
+    }
+
+    if (response.assets[x].name.slice(-9)=== "arm64.exe"){
```

Comment:
It may be easier to just use a for-of loop here. It's very similar to the range-based loop in c++. And use `endsWith()` for the substring checks, and maybe get fancy with some destructuring too:
```
for (const asset of response.assets) {
  const { name, browser_download_url: url } = asset;

  if ( !name || !url ) continue;

  if (name.endsWith("arm64.exe")) {
    buildMatrix.windows.arm64Installer.version = newVersion;
    buildMatrix.windows.arm64Installer.url = url;
  }
}
```

With the destructuring, "name" will be a copy of the similarly named properly of the "asset" object, and the value of "browser_download_url" will be assigned to a new variable named "url". If either of those properties are missing from the object then they will have a value of `undefined`.

Reply:
`endsWith` is a good change, because it is definitely a lot less janky.  But I don't think I want to go through the effort of redoing the loop, even if the javascript is better.

Reply:
Don't blame you. It can always be done with a future PR if the opportunity presents itself.